torch/distributions/categorical.py

```diff
@@ -73,11 +73,19 @@ def param_shape(self):
 
     @property
     def mean(self):
-        return self.probs.new([float('nan')]).expand(self._extended_shape())
+        # verbose because of differening Variable/Tensor apis and lack of dtypes
+        if isinstance(self.probs, Variable):
+            return variable(float('nan')).type_as(self.probs).expand(self._extended_shape())
```

Comment:
Does `.type_as()` place its result on the correct cuda device?

Reply:
apparently not.

Reply:
I always use `probs.new([float('nan')]).expand(...)`. What is the new syntax we can use for scalars to get correct device placement?